src/utils/middlewares/azure_user_attributes_manage.ts

```diff
@@ -0,0 +1,166 @@
+ *
+ * In **LEGACY flow** is used to retrieve the associated Service.
+ *
+ * In **MANAGE flow** is used to check if its name starts with "MANAGE-".
+ *
+ * On success, the middleware provides an IUserAttributes.
+ */
+export const AzureUserAttributesManageMiddleware = (
+  serviceModel?: ServiceModel
+): IRequestMiddleware<
+  | "IResponseErrorForbiddenNotAuthorized"
+  | "IResponseErrorQuery"
+  | "IResponseErrorInternal",
+  IAzureUserAttributes | IAzureUserAttributesManage
+> => async (
+  request
+): Promise<
+  E.Either<
+    IResponse<
+      | "IResponseErrorInternal"
+      | "IResponseErrorQuery"
+      | "IResponseErrorForbiddenNotAuthorized"
+    >,
+    IAzureUserAttributes | IAzureUserAttributesManage
+  >
+> => {
+  const errorOrUserEmail = EmailString.decode(
+    request.header(HEADER_USER_EMAIL)
+  );
+
+  if (E.isLeft(errorOrUserEmail)) {
+    return E.left<IResponse<"IResponseErrorInternal">, IAzureUserAttributes>(
+      ResponseErrorInternal(
+        `Missing, empty or invalid ${HEADER_USER_EMAIL} header`
+      )
+    );
+  }
+
+  const userEmail = errorOrUserEmail.right;
+
+  const errorOrUserSubscriptionId = NonEmptyString.decode(
+    request.header(HEADER_USER_SUBSCRIPTION_KEY)
+  );
+
+  if (E.isLeft(errorOrUserSubscriptionId)) {
+    return E.left<IResponse<"IResponseErrorInternal">, IAzureUserAttributes>(
+      ResponseErrorInternal(
+        `Missing or empty ${HEADER_USER_SUBSCRIPTION_KEY} header`
+      )
+    );
+  }
+
+  const subscriptionId = errorOrUserSubscriptionId.right;
+
+  if (!subscriptionId.startsWith("MANAGE-") && serviceModel) {
+    // LEGACY Flow
+    // serviceId equals subscriptionId
+    const errorOrMaybeService = await serviceModel.findLastVersionByModelId([
+      subscriptionId
+    ])();
+
+    if (E.isLeft(errorOrMaybeService)) {
+      winston.error(
+        `No service found for subscription|${subscriptionId}|${JSON.stringify(
+          errorOrMaybeService.left
+        )}`
+      );
+      return E.left<IResponse<"IResponseErrorQuery">, IAzureUserAttributes>(
+        ResponseErrorQuery(
+          `Error while retrieving the service tied to the provided subscription id`,
+          errorOrMaybeService.left
+        )
+      );
+    }
+
+    const maybeService = errorOrMaybeService.right;
+
+    if (isNone(maybeService)) {
+      winston.error(
+        `AzureUserAttributesMiddleware|Service not found|${subscriptionId}`
+      );
+      return E.left<
+        IResponse<"IResponseErrorForbiddenNotAuthorized">,
+        IAzureUserAttributes
+      >(ResponseErrorForbiddenNotAuthorized);
+    }
+
+    const authInfo: IAzureUserAttributes = {
+      email: userEmail,
+      kind: "IAzureUserAttributes",
+      service: maybeService.value
+    };
+
+    return E.right<
+      IResponse<
+        | "IResponseErrorForbiddenNotAuthorized"
+        | "IResponseErrorQuery"
+        | "IResponseErrorInternal"
+      >,
+      IAzureUserAttributes
+    >(authInfo);
```

Comment:
Unfortunately we cannot fully reference the `azure_user_attributes` module because within this module there are controls that would not work with the _**Manage**_ flow and the main goal we set ourselves was not to touch `azure_user_attributes` in order to avoid potential impacts on many APIs using this middleware.
For this reason we agreed to **temporarily duplicate** `azure_user_attributes` by adding an _"else"_ condition that would **temporarily** make both flows possible.
So summarizing, the answer is "yes, the new middleware duplicates `azure_user_attributes` by adding a condition (else) in which it also manages the MANAGE flow.